Fix encounter overview tables to display more than 14 items

Naman-45 opened the pull request:
## Proposed Changes

- Fixes #10869
- Increased the default table row limit from 14 to 100 for better visibility.
- Modified three files:
  - `src/components/Patient/allergy/list.tsx`
  - `src/components/Patient/symptoms/list.tsx`
  - `src/components/Patient/diagnosis/list.tsx`
- Attached relevant screenshots for reference.

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews.
- [ ] Completion of QA.

## Screenshots

![Screenshot 1](https://github.com/user-attachments/assets/5a9f6bf0-ed34-46a9-9bb2-c7af5765f477)  
![Screenshot 2](https://github.com/user-attachments/assets/9e45d05b-3aff-460a-9a8e-8ea1c3db4f36)  
![Screenshot 3](https://github.com/user-attachments/assets/39cfb1f8-e6f4-4b04-979a-b1a87af091ac)


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced a full-view encounter dialog to navigate between key sections such as allergies, symptoms, diagnoses, and consultation details.
  - Enhanced patient lists with pagination and refined filtering for a smoother viewing experience.
  - Updated interface texts with clearer labels and prompts for clinical history, assessments, examinations, and related areas.

- **Bug Fixes**
  - Improved filtering logic to exclude entries marked as "entered_in_error" for allergies and symptoms.

- **Chores**
  - Added constants for inactive clinical statuses related to allergies, symptoms, and diagnoses to improve status management.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Naman-45 committed:
Fix encounter overview tables to display more than 14 items

bodhish commented:
We should have a show all page for thse; 
<img width="730" alt="image" src="https://github.com/user-attachments/assets/d06a85df-f983-4250-b8f8-c04cad82c502" />


https://www.figma.com/design/Z93EYKSa1MdBmXndsMBJSQ/Care?node-id=7495-1049&t=7S0xTSX3GDDk0rxM-1

Naman-45 commented:
Just so that I get you right, are you asking me to add questionnaire response section the way right now there is allergy, diagnose and symptom section?

Jacobjeevan commented:
Converted to draft, mark as ready when you have committed the changes 👍

Naman-45 committed:
Add full view button and dialog box

Naman-45 committed:
Remove no longer needed hideFullViewButton from questionnaire list

Naman-45 commented:
I have added full view button and a dialog box

https://github.com/user-attachments/assets/ea7d9d56-683c-4d02-ac41-c4f0f550af14

Naman-45 committed:
Minor modifications

AdityaJ2305 commented:
@Naman-45 , Checks are failing

Naman-45 committed:
Merge branch 'develop' into issues/10869/fix-table-item-limits

Naman-45 committed:
Remove non nullish assertion

Naman-45 committed:
Merge branch 'issues/10869/fix-table-item-limits' of https://github.com/Naman-45/care_fe into issues/10869/fix-table-item-limits
Merge conflicts and resolves failing checks

Naman-45 commented:
> @Naman-45 , Checks are failing

Yes, working on it.

Naman-45 committed:
Resolve conflicts

Naman-45 committed:
Make facilityId optional

Naman-45 committed:
Remove facilityId completely

Naman-45 commented:
> @Naman-45 , Checks are failing

Now they are passing.

AdityaJ2305 commented:
@Naman-45 
<img width="848" alt="Screenshot 2025-03-07 at 11 27 13 PM" src="https://github.com/user-attachments/assets/d9d8e1c7-9953-4bbe-a51e-52dec38125c5" />

AdityaJ2305 reviewed (changes requested):
<img width="1154" alt="Screenshot 2025-03-07 at 11 30 31 PM" src="https://github.com/user-attachments/assets/b79bda75-9770-425c-a8ee-ec492200bcb4" />
You don't need to replicate exact data with design 😅

Naman-45 committed:
Debug Duplicate diagnoses List

Naman-45 commented:
> <img alt="Screenshot 2025-03-07 at 11 30 31 PM" width="1154" src="https://private-user-images.githubusercontent.com/148325158/420448467-b79bda75-9770-425c-a8ee-ec492200bcb4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDEzNzA4MTcsIm5iZiI6MTc0MTM3MDUxNywicGF0aCI6Ii8xNDgzMjUxNTgvNDIwNDQ4NDY3LWI3OWJkYTc1LTk3NzAtNDI1Yy1hOGVlLWVjNDkyMjAwYmNiNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMwN1QxODAxNTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NjZjMzI4NzIwZTIzM2E1Y2QyYzVmZmUyMTY5Y2Y3MGM1ODNjY2EwYmI3NWVhZmJkZWY0ZjFiZTI1YzA4ZTVkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.TSyXIugYMl7vJqSBt4lNFJiCU88fscJ774ymUXUU7pQ"> You don't need to replicate exact data with design 😅

I'm sorry, but I wasn't able to get date so for now I hardcoded it.

AdityaJ2305 commented on src/components/Patient/allergy/list.tsx:
```diff
@@ -85,13 +87,20 @@ export function AllergyList({
         encounter: completedEncounterStatus.includes(encounterStatus as string)
           ? encounterId
           : undefined,
+        limit: 100,
```
The whole purpose of using Full-Screen mode is to display items beyond the backend limit. Currently, it shows up to 100 items on both views

cc: @Jacobjeevan @rithviknishad

Naman-45 replied:
> The whole purpose of using Full-Screen mode is to display items beyond the backend limit. Currently, it shows up to 100 items on both views.

I think, since the default backend limit is only 14, it was limiting the total number of allergies, diagnoses and symptoms but now I think for these properties,100 limit is more than enough.

There are not even this many allergies, diagnoses and symptoms to add to the list.

AdityaJ2305 replied:
We don’t want to show more items than the backend limit in the overview section. Please check my previous comment one more time.

Naman-45 replied:
Oh my bad, I will fix it.

Naman-45 committed:
Set overview limit to default backend limit

AdityaJ2305 commented:
> > <img alt="Screenshot 2025-03-07 at 11 30 31 PM" width="1154" src="https://private-user-images.githubusercontent.com/148325158/420448467-b79bda75-9770-425c-a8ee-ec492200bcb4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDEzNzA4MTcsIm5iZiI6MTc0MTM3MDUxNywicGF0aCI6Ii8xNDgzMjUxNTgvNDIwNDQ4NDY3LWI3OWJkYTc1LTk3NzAtNDI1Yy1hOGVlLWVjNDkyMjAwYmNiNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMwN1QxODAxNTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NjZjMzI4NzIwZTIzM2E1Y2QyYzVmZmUyMTY5Y2Y3MGM1ODNjY2EwYmI3NWVhZmJkZWY0ZjFiZTI1YzA4ZTVkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.TSyXIugYMl7vJqSBt4lNFJiCU88fscJ774ymUXUU7pQ"> You don't need to replicate exact data with design 😅
> 
> I'm sorry, but I wasn't able to get date so for now I hardcoded it.
@Naman-45 , Remove this one

AdityaJ2305 commented on src/components/Patient/symptoms/list.tsx:
```diff
@@ -20,22 +23,31 @@ interface SymptomsListProps {
+  overviewSection = true,
 }: SymptomsListProps) {
   const [showEnteredInError, setShowEnteredInError] = useState(false);
+  let limit;
```
I don't think it is right way, use query.paginated 

let maintainers confirm it @amjithtitus09 @Jacobjeevan @rithviknishad

Naman-45 replied:
> I don't think it is right way, use query.paginated
> 
> let maintainers confirm it @amjithtitus09 @Jacobjeevan @rithviknishad

Should I wait or do it using query.paginated?

amjithtitus09 replied:
@Naman-45 @AdityaJ2305 Have a load more button after the records that loads the next set on click. Repeat until all are shown. Use Default limit (14)

AdityaJ2305 replied:
I think for this we need to use useInfiniteQuery right ? @amjithtitus09

Naman-45 committed:
Add date and time of assessment

Naman-45 commented:
> > > <img alt="Screenshot 2025-03-07 at 11 30 31 PM" width="1154" src="https://private-user-images.githubusercontent.com/148325158/420448467-b79bda75-9770-425c-a8ee-ec492200bcb4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDEzNzA4MTcsIm5iZiI6MTc0MTM3MDUxNywicGF0aCI6Ii8xNDgzMjUxNTgvNDIwNDQ4NDY3LWI3OWJkYTc1LTk3NzAtNDI1Yy1hOGVlLWVjNDkyMjAwYmNiNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzA3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMwN1QxODAxNTdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NjZjMzI4NzIwZTIzM2E1Y2QyYzVmZmUyMTY5Y2Y3MGM1ODNjY2EwYmI3NWVhZmJkZWY0ZjFiZTI1YzA4ZTVkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.TSyXIugYMl7vJqSBt4lNFJiCU88fscJ774ymUXUU7pQ"> You don't need to replicate exact data with design 😅
> > 
> > 
> > I'm sorry, but I wasn't able to get date so for now I hardcoded it.
> > @Naman-45 , Remove this one

Fixed it.

amjithtitus09 commented:
@Naman-45 Can you wire this popup to the existing "View all" button instead of a new Full View button? In the overview table, fetch records only from the active statuses and display first. Along with it, fire an API for inactive statuses with limit 0, to decide if "View all" button needs to be displayed or not. On click can open up the new Popup. Let me know if you need any clarifications.

Here is an image for reference (Credits: @AdityaJ2305 ). This would also fix https://github.com/ohcnetwork/care_fe/issues/11184
![image](https://github.com/user-attachments/assets/56779fd2-62c1-4245-bd29-75683d7d7221)

Naman-45 commented:
> @Naman-45 Can you wire this popup to the existing "View all" button instead of a new Full View button? In the overview table, fetch records only from the active statuses and display first. Along with it, fire an API for inactive statuses with limit 0, to decide if "View all" button needs to be displayed or not. On click can open up the new Popup. Let me know if you need any clarifications.
> 
> Here is an image for reference (Credits: @AdityaJ2305 ). This would also fix #11184 ![image](https://private-user-images.githubusercontent.com/37619282/420856703-56779fd2-62c1-4245-bd29-75683d7d7221.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE2MTkxMzgsIm5iZiI6MTc0MTYxODgzOCwicGF0aCI6Ii8zNzYxOTI4Mi80MjA4NTY3MDMtNTY3NzlmZDItNjJjMS00MjQ1LWJkMjktNzU2ODNkN2Q3MjIxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzEwVDE1MDAzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQxMWI3MTY5MzQwNjkzYjE0NWYxZDRkZDAwNzE0NGIxYzQxZjdiNjhhZGI0NjQyYmJiMTNkNzY2YmY5NTU4MzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.gxo1ULo_JLbCBPp7IMsaZNgD1SmC1AS-KxYnqdjdkwg)

Ok so I will remove Full View button and instead wire it up to the view all button? Now, as you said that you want to display all the active entries in the overview section so what all encounter statuses do you consider as active and which all inactive.

And In the overview section you want to display all the active statuses entries then what if they are more than 14 i.e; cannot be fetched in a single call, how do you want that case to be handled?

Jacobjeevan commented:
> > @Naman-45 Can you wire this popup to the existing "View all" button instead of a new Full View button? In the overview table, fetch records only from the active statuses and display first. Along with it, fire an API for inactive statuses with limit 0, to decide if "View all" button needs to be displayed or not. On click can open up the new Popup. Let me know if you need any clarifications.
> > Here is an image for reference (Credits: @AdityaJ2305 ). This would also fix #11184 ![image](https://private-user-images.githubusercontent.com/37619282/420856703-56779fd2-62c1-4245-bd29-75683d7d7221.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE2MTkxMzgsIm5iZiI6MTc0MTYxODgzOCwicGF0aCI6Ii8zNzYxOTI4Mi80MjA4NTY3MDMtNTY3NzlmZDItNjJjMS00MjQ1LWJkMjktNzU2ODNkN2Q3MjIxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzEwVDE1MDAzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQxMWI3MTY5MzQwNjkzYjE0NWYxZDRkZDAwNzE0NGIxYzQxZjdiNjhhZGI0NjQyYmJiMTNkNzY2YmY5NTU4MzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.gxo1ULo_JLbCBPp7IMsaZNgD1SmC1AS-KxYnqdjdkwg)
> 
> Ok so I will remove Full View button and instead wire it up to the view all button? Now, as you said that you want to display all the active entries in the overview section so what all encounter statuses do we consider as active and which all as inactive statuses.
> 
> And In the overview section as we want to display all the active entries then what if there are more than 14 i.e; cannot be fetched in a single call, how should I handle that case?



> > @Naman-45 Can you wire this popup to the existing "View all" button instead of a new Full View button? In the overview table, fetch records only from the active statuses and display first. Along with it, fire an API for inactive statuses with limit 0, to decide if "View all" button needs to be displayed or not. On click can open up the new Popup. Let me know if you need any clarifications.
> > Here is an image for reference (Credits: @AdityaJ2305 ). This would also fix #11184 ![image](https://private-user-images.githubusercontent.com/37619282/420856703-56779fd2-62c1-4245-bd29-75683d7d7221.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE2MTkxMzgsIm5iZiI6MTc0MTYxODgzOCwicGF0aCI6Ii8zNzYxOTI4Mi80MjA4NTY3MDMtNTY3NzlmZDItNjJjMS00MjQ1LWJkMjktNzU2ODNkN2Q3MjIxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzEwVDE1MDAzOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQxMWI3MTY5MzQwNjkzYjE0NWYxZDRkZDAwNzE0NGIxYzQxZjdiNjhhZGI0NjQyYmJiMTNkNzY2YmY5NTU4MzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.gxo1ULo_JLbCBPp7IMsaZNgD1SmC1AS-KxYnqdjdkwg)
> 
> Ok so I will remove Full View button and instead wire it up to the view all button? Now, as you said that you want to display all the active entries in the overview section so what all encounter statuses do we consider as active and which all as inactive statuses.
> 
> And In the overview section as we want to display all the active entries then what if there are more than 14 i.e; cannot be fetched in a single call, how should I handle that case?

Symptom and Diagnosis
- show everything else except:
   - clinical status: inactive, resolved
   - verification status: entered in error

Allergy
- show everything else except:
   - clinical status: inactive, resolved
   - verification status: entered in error

As for more than 14 entries, it's fine. Let's just show the default number of entries (14).

nihal467 commented:
![image](https://github.com/user-attachments/assets/471f1a31-938b-4d93-8b49-646698abc3fc)

- [ ] the close button is overlapping with the heading

Naman-45 commented:
> ![image](https://private-user-images.githubusercontent.com/57055998/421354966-471f1a31-938b-4d93-8b49-646698abc3fc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE3MTcyNDIsIm5iZiI6MTc0MTcxNjk0MiwicGF0aCI6Ii81NzA1NTk5OC80MjEzNTQ5NjYtNDcxZjFhMzEtOTM4Yi00ZDkzLThiNDktNjQ2Njk4YWJjM2ZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzExVDE4MTU0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTliMWEwMDljMmViYjdiNmRmNmE5ZDk3MTA3ZWU5ZDhmNGNiM2MyZDZmOGY4Y2NjZGEwNTkzNTNlMjIyZTA4MDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.QdsVK2H-O0ET-rVqIenvPyfUtuFHgvriHEGX0aAoK7E)
> 
> * [ ]  the close button is overlapping with the heading

![Screenshot From 2025-03-11 23-52-23](https://github.com/user-attachments/assets/1459a7d5-bf10-4061-b69c-e653a5129957)
I don't know whats happening, but there is no close button  showing in my screen.

AdityaJ2305 commented:
@Naman-45 , It is for mobile screen, you need to make the ui responsive

Naman-45 committed:
Remove Full view button and wire the popup to view all button

Naman-45 committed:
Add functionality to load more button

Naman-45 commented:
I think now I have committed all of the requested changes. I do have one confusion though, you asked me conditionally render view all button i.e; render only if there are any inactive entries and in the overview section we are only showing 14 active entries then what if there are more than 14 active entries and no inactive entries, how will those be accessible?

Naman-45 committed:
Add responsiveness in the dialog popup

Naman-45 committed:
Merge branch 'develop' into issues/10869/fix-table-item-limits

Naman-45 commented:
I had committed all of the requested changes, and its ready for review. cc @Jacobjeevan

Naman-45 commented:
@Jacobjeevan I had committed all of the requested changes, and its ready for review.

rithviknishad commented on src/components/Patient/diagnosis/list.tsx:
```diff
@@ -11,58 +11,97 @@ import { Button } from "@/components/ui/button";
+import { Encounter } from "@/types/emr/encounter";
 
 import { DiagnosisTable } from "./DiagnosisTable";
 
 interface DiagnosisListProps {
   patientId: string;
   encounterId?: string;
-  facilityId?: string;
   className?: string;
+  hideFullViewButton?: boolean;
+  encounter?: Encounter;
```
why pass both encounter and encounter id as a prop? 🤔 

encounter already has it's id inside the object

Naman-45 replied:
Oh yes, got it fixed.

Naman-45 committed:
Remove unnecessary props

Naman-45 committed:
Merge branch 'develop' into issues/10869/fix-table-item-limits

AdityaJ2305 reviewed (commented):
@Naman-45, we need to display the first 14 active items in the overview section tables. With the current implementation, it’s fetching 14 items in total, not specifically 14 active ones.
correct me if i'm wrong

Naman-45 commented:
> @Naman-45, we need to display the first 14 active items in the overview section tables. With the current implementation, it’s fetching 14 items in total, not specifically 14 active ones. correct me if i'm wrong

We are only displaying active entries only in the overview section, as you can see in the below screen shot, by default i.e; in the overview section hideFullViewButton is false and when the dialog pops up hideFullViewButton prop is passed as true.
![Screenshot From 2025-03-20 02-14-31](https://github.com/user-attachments/assets/cdca52a9-b1c6-43de-a5c3-7074cbd81b21)

Naman-45 committed:
Better nomenclature for dialog popup prop

Naman-45 commented:
> > @Naman-45, we need to display the first 14 active items in the overview section tables. With the current implementation, it’s fetching 14 items in total, not specifically 14 active ones. correct me if i'm wrong
> 
> We are displaying only active entries in the overview section, as you can see in the below screen shot, by default i.e; in the overview section hideFullViewButton is false and when the dialog pops up hideFullViewButton prop is passed as true. ![Screenshot From 2025-03-20 02-14-31](https://private-user-images.githubusercontent.com/126798758/424690637-cdca52a9-b1c6-43de-a5c3-7074cbd81b21.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDI0MTc2MDIsIm5iZiI6MTc0MjQxNzMwMiwicGF0aCI6Ii8xMjY3OTg3NTgvNDI0NjkwNjM3LWNkY2E1MmE5LWIxYzYtNDNkZS1hNWMzLTcwNzRjYmQ4MWIyMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMzE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDMxOVQyMDQ4MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hODNmODE0MWJmYTE4NjdmZGE2NDU4MGQ4ZTQwODk0NDBkNzk3YjkwOThhZGZkNzlkOWUwOGQwZTE5MGZlNGExJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.sOuhjmHpYfNn3CWNKdOzvUU2BWL-wcFrSKORvu3HUgA)

Also change previous nomenclature of dialog popup prop from hideFullViewButton -> dialogView

AdityaJ2305 commented:
You’re filtering from the first 14 items, right? That won’t guarantee the first 14 active ones. For example, if the fetch returns 10 active and 4 entered_in_error, you’ll only display those 10 active ones, even though more active items exist but weren’t included in the initial request.

Naman-45 commented:
> You’re filtering from the first 14 items, right? That won’t guarantee the first 14 active ones. For example, if the fetch returns 10 active and 4 entered_in_error, you’ll only display those 10 active ones, even though more active items exist but weren’t included in the initial request.

I also have the same doubt - 

https://github.com/ohcnetwork/care_fe/pull/10871/#issuecomment-2722059672

Naman-45 commented:
> You’re filtering from the first 14 items, right? That won’t guarantee the first 14 active ones. For example, if the fetch returns 10 active and 4 entered_in_error, you’ll only display those 10 active ones, even though more active items exist but weren’t included in the initial request.

I also have the same doubt, hence I asked the following

![Screenshot From 2025-03-20 02-49-45](https://github.com/user-attachments/assets/1b20e6cb-0589-4081-b2a5-91c9d519ec0e)

and @Jacobjeevan asked to display only 14 entries..

![Screenshot From 2025-03-20 02-50-12](https://github.com/user-attachments/assets/5e5dfdc2-0083-43d8-ae16-10c658647e73)

![Screenshot From 2025-03-20 02-51-46](https://github.com/user-attachments/assets/e3ed4692-d8db-4aef-b352-bcca6ada56f4)

Jacobjeevan commented:
> I think now I have committed all of the requested changes. I do have one confusion though, you asked me to conditionally render view all button i.e; render only if there are any inactive entries and in the overview section we are only showing 14 active entries then what if there are more than 14 active entries and no inactive entries, how will those be accessible?

The dialog should show all entries; backend supports the filter, so you can make sure to get 14 active ones.

https://careapi.ohc.network/swagger/#/allergy-intolerance/api_v1_patient_allergy_intolerance_list
https://careapi.ohc.network/swagger/#/diagnosis/api_v1_patient_diagnosis_list
https://careapi.ohc.network/swagger/#/symptom/api_v1_patient_symptom_list

Naman-45 committed:
Add filtering through backend using query params

Naman-45 committed:
Correct query key name

Naman-45 committed:
Add dependency array in useEffect

Naman-45 committed:
Resolve merge conflicts and add more qery params to optimize filtering through backend

Naman-45 commented:
@Jacobjeevan  Ready for review, Following is the summary of what I have done-

-Instead of filtering client side I am now filtering through backend using query params.
-Converted dialogView into a state variable
-Added onClose prop in the fullVIewDialog component through which when we close the dialog we can again set the dialogView state to false.
-Instead of filtering and checking for any inactive on the client side, fetching and checking that too using backend filtering.
-Excluded entered_in_error through  exclude_verification_status query param, making sure we get all the actual entries.


PS:  After resolving the merge conflicts I'm unable to run the app because of some errors in the following files-

src/components/ui/calender
src/components/questionnaire/QuestionTypes/DateQuestion

Naman-45 committed:
Remove duplicate Diagnosis Table

Naman-45 committed:
Fix some inactive clinical status lists

Commit:
Merge branch 'develop' into issues/10869/fix-table-item-limits

bodhish commented:
@Naman-45 this doesn't address the issue correctly, The UI was supposed to be a sheet that we can call from any point that has an option to navigate between each of these. I will take this over.
